app/components/channel_drawer/channels_list/channel_item/index.js

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {getCurrentChannelId, getChannel} from 'mattermost-redux/selectors/entities/channels';
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import ChannelItem from './channel_item';
+
+function mapStateToProps(state, ownProps) {
+    const channel = getChannel(state, ownProps.channelId);
+
+    return {
+        currentChannelId: getCurrentChannelId(state),
+        displayName: channel.display_name,
```

Comment:
How does this work for direct and group messages? Doesn't getChannel return the channel without those set?

Reply:
getChannel is using the `completeDirectInfo` stuff

Reply:
Oh, in that case, this isn't memoizing at all since the channel ID is changing constantly so we're getting different objects each time. We should use `makeGetChannel` here and get rid of the plain `getChannel` selector since it's doing that